### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by ArbitRandomUser**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [ ] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [ ] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [ ] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [ ] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [ ] Did I properly add test dependencies to the `test` directory (if applicable)?
- [ ] Did I check relevant tutorials that may be affected by changes in this PR?
- [ ] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #

**How did you address these issues with this PR? What methods did you use?**

Rudimentary PR on arrange objects, will add details soon ...

### CODE REVIEW

File: src/arrange.jl [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                push!(ps, Point((m * [p.x, p.y, 1])[1:2]...))
+            end
+        end
+    end
+    Luxor.BoundingBox(ps)
+end
+
+
+"""
+    arrange()
```
Comment (Sov-trotter) [inline src/arrange.jl:91]: You can add the parameters here

File: src/arrange.jl [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+arranges objects 
+returns a closure to be used with act!(frames,Function)
+
+frames:: the global frames  during which the arrangment should take place
+objects:: array of objects that should be aligned
+p:: Point under/to the side of which arranging should take place
```
Comment (Sov-trotter) [inline src/arrange.jl:98]: This can be elaborated further. Maybe take the example of  `0+10`

**Comment by ArbitRandomUser** [comment]
@Sov-trotter @Wikunia @gpucce @TheCedarPrince 
Ive added some more features , to align objects in variuos directions 
around a point.
```julia
using Javis
video = Video(1000,1000)
nframes=250
groundj = Background(1:nframes,(args...) -> begin
               background("black")
               sethue("white")
           end)
objj1 = Object(1:nframes,(args...)-> begin
                  box(O,100,100,:stroke)
                  return O
                  end,
                  Point(100,150)
                 )

objj2 = Object(1:nframes,(args...)-> begin
                  circle(O,50,:stroke)
                  return O
                  end,
                  Point(-100,250)
                 )

objj3 = Object(1:nframes,(args...)-> begin
                  #rotate(2π/12)
                  box(O,100,50,:stroke)
                  #star(O,50,3,0.5,0.0,:stroke)
                  return O
                  end,
                  Point(-200,-150)
                 )


objlist = [objj1,objj2,objj3]
frameno = 2
Object(1:nframes,(args...)->circle(O,5,:fill))
#act!(frameno,arrange(frameno:frameno+25,objlist,O;gap=0,dir=:horizontal,align=:left,towards=:top))
for dir in (:horizontal,:vertical)
    global frameno
    for align in (:left,:right)
        for towards in (:top,:bottom)
            act!(frameno,arrange(frameno:frameno+25,objlist,O;gap=0,dir=dir,align=align,towards=towards))
            frameno=frameno+30
        end
    end
end

render(video,pathname="testarrange3.gif")
run(`mpv testarrange3.gif`)

```
testarrange3

File: src/runtimefuncs.jl [hunk 3]
```diff
@@ -0,0 +1,36 @@
+"""
+a dictionary mapping frameno => Array{Func}
+the functions should take 1 argument the frame number
+
+at render time RuntimeFunctionMap[frame] is checked and every Func 
+in the array is called with `func(frame)`
+"""
+#videoRuntimeFunctionDict = Dict{Int,Array{Function}}()
```
Comment (Sov-trotter) [inline src/runtimefuncs.jl:8]: Is this still needed?

**Comment by ArbitRandomUser** [comment]
P.R also adds ability to arrange around objects now 
```julia
using Javis
video = Video(1000,1000)
nframes=300
groundj = Background(1:nframes,(args...) -> begin
               background("black")
               sethue("white")
           end)
objj1 = Object(1:nframes,(args...)-> begin
                  sethue("yellow")
                  box(O,100,100,:stroke)
                  text("Target")
                  return O
                  end,
                  O-100
                 )

objj2 = Object(1:nframes,(args...)-> begin
                  circle(O,35,:stroke)
                  return O
                  end,
                  Point(-100,250)
                 )

objj3 = Object(1:nframes,(args...)-> begin
                  #rotate(2π/12)
                  box(O,150,50,:stroke)
                  #star(O,50,3,0.5,0.0,:stroke)
                  return O
                  end,
                  Point(-200,-150)
                 )


objlist = [objj2,objj3]
frameno = 20
for dir in (:horizontal,:vertical)
    global frameno
    for halign in (:left,:right)
        for valign in (:top,:bottom)
            act!(frameno,arrange(frameno:frameno+25,objlist,objj1;gap=0,dir=dir,halign=halign,valign=valign))
            frameno=frameno+30
        end
    end
end

render(video,pathname="testarrange_obj.gif")
run(`mpv testarrange_obj.gif`)
```
testarrange_obj

File: src/arrange.jl (hunk 2 above)
Comment (ArbitRandomUser) [reply src/arrange.jl:186]: didnt quite catch what you wanted , i forgot maybe i added the example that folows after seeing your comment , is that what you wanted

Comment (ArbitRandomUser) [reply src/arrange.jl:186]: okay it looks like i did add after seeing your comment, couldnt see the entire changes on this page itself . Take a look and unresolv/leave new comment if its not what you were talking about
